### PR TITLE
Add mutexes to enable multithreading safe timer operation

### PR DESCRIPTION
**Pull request by uschwes**

### CODE REVIEW

File: sm_timing/src/Timer.cpp [hunk 1]
```diff
@@ -32,6 +34,7 @@ namespace timing {
     if(i == instance().m_tagMap.end()) {
       // If it is not there, create a tag.
       size_t handle =  instance().m_timers.size();
+      boost::mutex::scoped_lock lock(m_mutex);
```
Comment (HannesSommer) [inline sm_timing/src/Timer.cpp]: This line must move up one line.

File: sm_timing/src/Timer.cpp [hunk 2]
```diff
@@ -88,6 +91,7 @@ namespace timing {
   
   void Timer::start(){
     SM_ASSERT_TRUE(TimerException,!m_timing,"The timer " + Timing::getTag(m_handle) + " is already running");
+    boost::mutex::scoped_lock lock(m_mutex);
```
Comment (HannesSommer) [inline sm_timing/src/Timer.cpp]: Timer does not need a mutex at all.

File: sm_timing/src/Timer.cpp (hunk 1 above)
Comment (HannesSommer) [inline sm_timing/src/Timer.cpp]: No, even more. Up to the very beginning.

File: sm_timing/src/Timer.cpp [hunk 3]
```diff
@@ -179,6 +185,7 @@ namespace timing {
 
   void Timing::reset(size_t handle) {
     SM_ASSERT_LT(TimerException, handle, instance().m_timers.size(), "Handle is out of range: " << handle << ", number of timers: " << instance().m_timers.size());
+    boost::mutex::scoped_lock lock(m_mutex);
```
Comment (HannesSommer) [inline sm_timing/src/Timer.cpp]: Needs to be moved up.

**Comment by uschwes** [comment]
Builds on ubuntu, merge ok.

**Pull request merged**